### PR TITLE
fix: add env vars from sfdx-cli init to envvars

### PR DESCRIPTION
**Pull request by peternhale**
@W-10536395@

### CODE REVIEW

**Comment by peternhale** [comment]
NOTE: While simply copy/paste/fix of sfdx-cli env var init logic into salesfirlce/cli was possible, I made the lion's share of the changes in core.
- Add env vars not present in well=known env vars
- Add the init functions from sfdx-cli to core.

File: src/config/envVars.ts [hunk 1]
```diff
@@ -368,6 +430,69 @@ export class EnvVars extends Env {
     }, new Map<string, string>());
   }
 
+  public isAutoupdateDisabled(): boolean {
```
Comment (peternhale) [inline src/config/envVars.ts:433]: Function from sfdx-cli util/env.ts

File: src/globalInfo/whichCli.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export interface WhichCliInterface {
```
Comment (peternhale) [inline src/globalInfo/whichCli.ts:8]: Forward looking to when env var scopes will needed to be resolved.

**Review by mdonnalley — CHANGES_REQUESTED** [review]

**Review by RodEsp — APPROVED** [review]

**Comment by RodEsp** [comment]
@peternhale I thought there was still open discussion on this [in slack](https://salesforce-internal.slack.com/archives/C02J2118Z47/p1644607850992339?thread_ts=1644605732.525139&cid=C02J2118Z47). Why'd you go ahead and merge it?

**Pull request merged**